// @openzeppelin/contracts/build/contracts/Checkpoints__factory.ts
/* Autogenerated file. Do not edit manually. */
/* tslint:disable */
/* eslint-disable */
import {
  Contract,
  ContractFactory,
  ContractTransactionResponse,
  Interface,
} from "ethers";
import type { Signer, ContractDeployTransaction, ContractRunner } from "ethers";
import type { NonPayableOverrides } from "../../../../../../common";
import type {
  Checkpoints,
  CheckpointsInterface,
} from "../../../../../../node_modules/@openzeppelin/contracts/build/contracts/Checkpoints";

const _abi = [
  {
    inputs: [],
    name: "CheckpointUnorderedInsertion",
    type: "error",
  },
] as const;

const _bytecode =
  "0x60556032600b8282823980515f1a607314602657634e487b7160e01b5f525f60045260245ffd5b305f52607381538281f3fe730000000000000000000000000000000000000000301460806040525f80fdfea26469706673582212206ab76338e2e30aaa09c27eb4c7d8a4f399844190c598c55c2f00bc1e504e8c3164736f6c63430008180033";

type CheckpointsConstructorParams =
  | [signer?: Signer]
  | ConstructorParameters<typeof ContractFactory>;

const isSuperArgs = (
  xs: CheckpointsConstructorParams
): xs is ConstructorParameters<typeof ContractFactory> => xs.length > 1;

export class Checkpoints__factory extends ContractFactory {
  constructor(...args: CheckpointsConstructorParams) {
    if (isSuperArgs(args)) {
      super(...args);
    } else {
      super(_abi, _bytecode, args[0]);
    }
  }

  override getDeployTransaction(
    overrides?: NonPayableOverrides & { from?: string }
  ): Promise<ContractDeployTransaction> {
    return super.getDeployTransaction(overrides || {});
  }
  override deploy(overrides?: NonPayableOverrides & { from?: string }) {
    return super.deploy(overrides || {}) as Promise<
      Checkpoints & {
        deploymentTransaction(): ContractTransactionResponse;
      }
    >;
  }
  override connect(runner: ContractRunner | null): Checkpoints__factory {
    return super.connect(runner) as Checkpoints__factory;
  }

  static readonly bytecode = _bytecode;
  static readonly abi = _abi;
  static createInterface(): CheckpointsInterface {
    return new Interface(_abi) as CheckpointsInterface;
  }
  static connect(address: string, runner?: ContractRunner | null): Checkpoints {
    return new Contract(address, _abi, runner) as unknown as Checkpoints;
  }
}
