// @openzeppelin/contracts/build/contracts/IEntryPointExtra__factory.ts
/* Autogenerated file. Do not edit manually. */
/* tslint:disable */
/* eslint-disable */

import { Contract, Interface, type ContractRunner } from "ethers";
import type {
  IEntryPointExtra,
  IEntryPointExtraInterface,
} from "../../../../../../node_modules/@openzeppelin/contracts/build/contracts/IEntryPointExtra";

const _abi = [
  {
    inputs: [
      {
        components: [
          {
            internalType: "address",
            name: "sender",
            type: "address",
          },
          {
            internalType: "uint256",
            name: "nonce",
            type: "uint256",
          },
          {
            internalType: "bytes",
            name: "initCode",
            type: "bytes",
          },
          {
            internalType: "bytes",
            name: "callData",
            type: "bytes",
          },
          {
            internalType: "bytes32",
            name: "accountGasLimits",
            type: "bytes32",
          },
          {
            internalType: "uint256",
            name: "preVerificationGas",
            type: "uint256",
          },
          {
            internalType: "bytes32",
            name: "gasFees",
            type: "bytes32",
          },
          {
            internalType: "bytes",
            name: "paymasterAndData",
            type: "bytes",
          },
          {
            internalType: "bytes",
            name: "signature",
            type: "bytes",
          },
        ],
        internalType: "struct PackedUserOperation",
        name: "userOp",
        type: "tuple",
      },
    ],
    name: "getUserOpHash",
    outputs: [
      {
        internalType: "bytes32",
        name: "",
        type: "bytes32",
      },
    ],
    stateMutability: "view",
    type: "function",
  },
] as const;

export class IEntryPointExtra__factory {
  static readonly abi = _abi;
  static createInterface(): IEntryPointExtraInterface {
    return new Interface(_abi) as IEntryPointExtraInterface;
  }
  static connect(
    address: string,
    runner?: ContractRunner | null
  ): IEntryPointExtra {
    return new Contract(address, _abi, runner) as unknown as IEntryPointExtra;
  }
}
