// @openzeppelin/contracts/build/contracts/MerkleProof__factory.ts
/* Autogenerated file. Do not edit manually. */
/* tslint:disable */
/* eslint-disable */
import {
  Contract,
  ContractFactory,
  ContractTransactionResponse,
  Interface,
} from "ethers";
import type { Signer, ContractDeployTransaction, ContractRunner } from "ethers";
import type { NonPayableOverrides } from "../../../../../../common";
import type {
  MerkleProof,
  MerkleProofInterface,
} from "../../../../../../node_modules/@openzeppelin/contracts/build/contracts/MerkleProof";

const _abi = [
  {
    inputs: [],
    name: "MerkleProofInvalidMultiproof",
    type: "error",
  },
] as const;

const _bytecode =
  "0x60556032600b8282823980515f1a607314602657634e487b7160e01b5f525f60045260245ffd5b305f52607381538281f3fe730000000000000000000000000000000000000000301460806040525f80fdfea26469706673582212204111738e32f018dd82f3e63441ffe7a8370ce18b75c54dfda73f3631a414f0f064736f6c63430008180033";

type MerkleProofConstructorParams =
  | [signer?: Signer]
  | ConstructorParameters<typeof ContractFactory>;

const isSuperArgs = (
  xs: MerkleProofConstructorParams
): xs is ConstructorParameters<typeof ContractFactory> => xs.length > 1;

export class MerkleProof__factory extends ContractFactory {
  constructor(...args: MerkleProofConstructorParams) {
    if (isSuperArgs(args)) {
      super(...args);
    } else {
      super(_abi, _bytecode, args[0]);
    }
  }

  override getDeployTransaction(
    overrides?: NonPayableOverrides & { from?: string }
  ): Promise<ContractDeployTransaction> {
    return super.getDeployTransaction(overrides || {});
  }
  override deploy(overrides?: NonPayableOverrides & { from?: string }) {
    return super.deploy(overrides || {}) as Promise<
      MerkleProof & {
        deploymentTransaction(): ContractTransactionResponse;
      }
    >;
  }
  override connect(runner: ContractRunner | null): MerkleProof__factory {
    return super.connect(runner) as MerkleProof__factory;
  }

  static readonly bytecode = _bytecode;
  static readonly abi = _abi;
  static createInterface(): MerkleProofInterface {
    return new Interface(_abi) as MerkleProofInterface;
  }
  static connect(address: string, runner?: ContractRunner | null): MerkleProof {
    return new Contract(address, _abi, runner) as unknown as MerkleProof;
  }
}
