// @openzeppelin/contracts/build/contracts/NoncesKeyed__factory.ts
/* Autogenerated file. Do not edit manually. */
/* tslint:disable */
/* eslint-disable */

import { Contract, Interface, type ContractRunner } from "ethers";
import type {
  NoncesKeyed,
  NoncesKeyedInterface,
} from "../../../../../../node_modules/@openzeppelin/contracts/build/contracts/NoncesKeyed";

const _abi = [
  {
    inputs: [
      {
        internalType: "address",
        name: "account",
        type: "address",
      },
      {
        internalType: "uint256",
        name: "currentNonce",
        type: "uint256",
      },
    ],
    name: "InvalidAccountNonce",
    type: "error",
  },
  {
    inputs: [
      {
        internalType: "address",
        name: "owner",
        type: "address",
      },
    ],
    name: "nonces",
    outputs: [
      {
        internalType: "uint256",
        name: "",
        type: "uint256",
      },
    ],
    stateMutability: "view",
    type: "function",
  },
  {
    inputs: [
      {
        internalType: "address",
        name: "owner",
        type: "address",
      },
      {
        internalType: "uint192",
        name: "key",
        type: "uint192",
      },
    ],
    name: "nonces",
    outputs: [
      {
        internalType: "uint256",
        name: "",
        type: "uint256",
      },
    ],
    stateMutability: "view",
    type: "function",
  },
] as const;

export class NoncesKeyed__factory {
  static readonly abi = _abi;
  static createInterface(): NoncesKeyedInterface {
    return new Interface(_abi) as NoncesKeyedInterface;
  }
  static connect(address: string, runner?: ContractRunner | null): NoncesKeyed {
    return new Contract(address, _abi, runner) as unknown as NoncesKeyed;
  }
}
